File: src/network/packet/data/LogType.php
<?php

namespace pocketcloud\cloud\network\packet\data;

use pocketcloud\cloud\console\log\level\CloudLogLevel;
use pocketcloud\cloud\util\misc\Writeable;
use pocketcloud\cloud\util\trait\EnumHelperTrait;

enum LogType implements Writeable {
    use EnumHelperTrait;

    case INFO;
    case WARN;
    case ERROR;
    case SUCCESS;
    case DEBUG;

    public function getName(): string {
        return $this->name;
    }

    public function toLogLevel(): CloudLogLevel {
        return match ($this) {
            self::INFO => CloudLogLevel::INFO(),
            self::WARN => CloudLogLevel::WARN(),
            self::ERROR => CloudLogLevel::ERROR(),
            self::SUCCESS => CloudLogLevel::SUCCESS(),
            self::DEBUG => CloudLogLevel::DEBUG()
        };
    }

    public function write(): string {
        return $this->name;
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/type/LogType.java
package de.pocketcloud.cloud.network.packet.type;

import de.pocketcloud.cloud.console.log.CloudLogLevel;
import de.pocketcloud.cloud.util.Writable;

public enum LogType implements Writable<String> {

    INFO,
    WARN,
    ERROR,
    SUCCESS,
    DEBUG;

    public CloudLogLevel toLogLevel() {
        return switch (this) {
            case INFO -> CloudLogLevel.INFO;
            case WARN -> CloudLogLevel.WARN;
            case ERROR -> CloudLogLevel.ERROR;
            case SUCCESS -> CloudLogLevel.SUCCESS;
            case DEBUG -> CloudLogLevel.DEBUG;
        };
    }

    @Override
    public String write() {
        return name();
    }
}